-Wmaybe-unitialized warnings in RenderingDeviceVulkan with GCC

**Godot version:**
4.0.r1.0187cda-1
**OS/device including version:**
arch linux intel 4500u
built with gcc version 10.1.0
**Issue description:**
While compiling:
```
[Initial build] Compiling ==> platform/linuxbsd/export/export.cpp
In file included from ./core/ustring.h:35,
                 from ./core/string_name.h:36,
                 from ./core/node_path.h:34,
                 from ./core/hashfuncs.h:36,
                 from ./core/oa_hash_map.h:34,
                 from drivers/vulkan/rendering_device_vulkan.h:34,
                 from drivers/vulkan/rendering_device_vulkan.cpp:31:
./core/cowdata.h: In member function 'VkRenderPass_T* RenderingDeviceVulkan::_render_pass_create(const Vector<RenderingDevice::AttachmentFormat>&, RenderingDevice::InitialAction, RenderingDevice::FinalAction, RenderingDevice::InitialAction, RenderingDevice::FinalAction, int*)':
./core/cowdata.h:141:3: error: '*((void*)& description +32)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  141 |   _get_data()[p_index] = p_elem;
      |   ^~~~~~~~~
drivers/vulkan/rendering_device_vulkan.cpp:2994:27: note: '*((void*)& description +32)' was declared here
 2994 |   VkAttachmentDescription description;
      |                           ^~~~~~~~~~~
In file included from ./core/ustring.h:35,
                 from ./core/string_name.h:36,
                 from ./core/node_path.h:34,
                 from ./core/hashfuncs.h:36,
                 from ./core/oa_hash_map.h:34,
                 from drivers/vulkan/rendering_device_vulkan.h:34,
                 from drivers/vulkan/rendering_device_vulkan.cpp:31:
./core/cowdata.h:141:3: error: '*((void*)& description +24)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  141 |   _get_data()[p_index] = p_elem;
      |   ^~~~~~~~~
drivers/vulkan/rendering_device_vulkan.cpp:2994:27: note: '*((void*)& description +24)' was declared here
 2994 |   VkAttachmentDescription description;
      |                           ^~~~~~~~~~~
In file included from ./core/ustring.h:35,
                 from ./core/string_name.h:36,
                 from ./core/node_path.h:34,
                 from ./core/hashfuncs.h:36,
                 from ./core/oa_hash_map.h:34,
                 from drivers/vulkan/rendering_device_vulkan.h:34,
                 from drivers/vulkan/rendering_device_vulkan.cpp:31:
./core/cowdata.h:141:3: error: '*((void*)& description +16)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  141 |   _get_data()[p_index] = p_elem;
      |   ^~~~~~~~~
drivers/vulkan/rendering_device_vulkan.cpp:2994:27: note: '*((void*)& description +16)' was declared here
 2994 |   VkAttachmentDescription description;
      |                           ^~~~~~~~~~~
[Initial build] Compiling ==> platform/osx/export/export.cpp
[Initial build] Compiling ==> platform/uwp/export/export.cpp
In file included from ./core/ustring.h:35,
                 from ./core/string_name.h:36,
                 from ./core/node_path.h:34,
                 from ./core/hashfuncs.h:36,
                 from ./core/oa_hash_map.h:34,
                 from drivers/vulkan/rendering_device_vulkan.h:34,
                 from drivers/vulkan/rendering_device_vulkan.cpp:31:
./core/cowdata.h: In member function 'void RenderingDeviceVulkan::_draw_list_insert_clear_region(RenderingDeviceVulkan::DrawList*, RenderingDeviceVulkan::Framebuffer*, Point2i, Point2i, bool, const Vector<Color>&, bool, float, uint32_t)':
./core/cowdata.h:141:3: error: 'clear_at.VkClearAttachment::clearValue.VkClearValue::color.VkClearColorValue::float32[3]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  141 |   _get_data()[p_index] = p_elem;
      |   ^~~~~~~~~
drivers/vulkan/rendering_device_vulkan.cpp:5543:21: note: 'clear_at.VkClearAttachment::clearValue.VkClearValue::color.VkClearColorValue::float32[3]' was declared here
 5543 |   VkClearAttachment clear_at;
      |                     ^~~~~~~~
In file included from ./core/ustring.h:35,
                 from ./core/string_name.h:36,
                 from ./core/node_path.h:34,
                 from ./core/hashfuncs.h:36,
                 from ./core/oa_hash_map.h:34,
                 from drivers/vulkan/rendering_device_vulkan.h:34,
                 from drivers/vulkan/rendering_device_vulkan.cpp:31:
./core/cowdata.h:141:3: error: 'clear_at.VkClearAttachment::clearValue.VkClearValue::color.VkClearColorValue::float32[2]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  141 |   _get_data()[p_index] = p_elem;
      |   ^~~~~~~~~
drivers/vulkan/rendering_device_vulkan.cpp:5543:21: note: 'clear_at.VkClearAttachment::clearValue.VkClearValue::color.VkClearColorValue::float32[2]' was declared here
 5543 |   VkClearAttachment clear_at;
      |                     ^~~~~~~~
cc1plus: all warnings being treated as errors
scons: *** [drivers/vulkan/rendering_device_vulkan.linuxbsd.opt.tools.64.o] Error 1
scons: building terminated because of errors.
==> ERROR: A failure occurred in build().
    Aborting...
```

--- a/drivers/vulkan/rendering_device_vulkan.cpp
+++ b/drivers/vulkan/rendering_device_vulkan.cpp
@@ -2991,16 +2991,21 @@ VkRenderPass RenderingDeviceVulkan::_render_pass_create(const Vector<AttachmentF
 	Vector<VkAttachmentReference> resolve_references;
 
 	for (int i = 0; i < p_format.size(); i++) {
-		VkAttachmentDescription description;
-
-		description.flags = 0;
 		ERR_FAIL_INDEX_V(p_format[i].format, DATA_FORMAT_MAX, VK_NULL_HANDLE);
-		description.format = vulkan_formats[p_format[i].format];
 		ERR_FAIL_INDEX_V(p_format[i].samples, TEXTURE_SAMPLES_MAX, VK_NULL_HANDLE);
+		ERR_FAIL_COND_V_MSG(!(p_format[i].usage_flags & (TEXTURE_USAGE_COLOR_ATTACHMENT_BIT | TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT | TEXTURE_USAGE_RESOLVE_ATTACHMENT_BIT)),
+				VK_NULL_HANDLE, "Texture format for index (" + itos(i) + ") requires an attachment (depth, stencil or resolve) bit set.");
+
+		VkAttachmentDescription description;
+		description.flags = 0;
+		description.format = vulkan_formats[p_format[i].format];
 		description.samples = rasterization_sample_count[p_format[i].samples];
-		//anything below does not really matter, as vulkan just ignores it when creating a pipeline
-		ERR_FAIL_COND_V_MSG(!(p_format[i].usage_flags & (TEXTURE_USAGE_COLOR_ATTACHMENT_BIT | TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT | TEXTURE_USAGE_RESOLVE_ATTACHMENT_BIT)), VK_NULL_HANDLE,
-				"Texture format for index (" + itos(i) + ") requires an attachment (depth, stencil or resolve) bit set.");
+		description.loadOp = VK_ATTACHMENT_LOAD_OP_MAX_ENUM; // Invalid value.
+		description.storeOp = VK_ATTACHMENT_STORE_OP_MAX_ENUM; // Invalid value.
+		description.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_MAX_ENUM; // Invalid value.
+		description.stencilStoreOp = VK_ATTACHMENT_STORE_OP_MAX_ENUM; // Invalid value.
+		description.initialLayout = VK_IMAGE_LAYOUT_MAX_ENUM; // Invalid value.
+		description.finalLayout = VK_IMAGE_LAYOUT_MAX_ENUM; // Invalid value.
 
 		bool is_depth_stencil = p_format[i].usage_flags & TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
 		bool is_sampled = p_format[i].usage_flags & TEXTURE_USAGE_SAMPLING_BIT;
@@ -3113,6 +3118,15 @@ VkRenderPass RenderingDeviceVulkan::_render_pass_create(const Vector<AttachmentF
 				ERR_FAIL_V(VK_NULL_HANDLE); //should never reach here
 			}
 		}
+
+		// Ensure VkAttachmentDescription has been initialized properly.
+		ERR_CONTINUE_MSG(description.loadOp == VK_ATTACHMENT_LOAD_OP_MAX_ENUM ||
+								 description.storeOp == VK_ATTACHMENT_STORE_OP_MAX_ENUM ||
+								 description.stencilLoadOp == VK_ATTACHMENT_LOAD_OP_MAX_ENUM ||
+								 description.stencilStoreOp == VK_ATTACHMENT_STORE_OP_MAX_ENUM ||
+								 description.initialLayout == VK_IMAGE_LAYOUT_MAX_ENUM ||
+								 description.finalLayout == VK_IMAGE_LAYOUT_MAX_ENUM,
+				"Bug: VkAttachmentDescription not initialized properly.");
 
 		attachments.push_back(description);
 
@@ -5541,6 +5555,10 @@ void RenderingDeviceVulkan::_draw_list_insert_clear_region(DrawList *draw_list, 
 	for (int i = 0; i < framebuffer->texture_ids.size(); i++) {
 		Texture *texture = texture_owner.getornull(framebuffer->texture_ids[i]);
 		VkClearAttachment clear_at;
+		clear_at.aspectMask = VK_IMAGE_ASPECT_FLAG_BITS_MAX_ENUM; // Invalid value.
+		clear_at.colorAttachment = uint32_t(-1); // Invalid value.
+		clear_at.clearValue.depthStencil.stencil = uint32_t(-1); // Invalid value.
+
 		if (p_clear_color && texture->usage_flags & TEXTURE_USAGE_COLOR_ATTACHMENT_BIT) {
 			ERR_FAIL_INDEX(color_index, p_clear_colors.size()); //a bug
 			Color clear_color = p_clear_colors[color_index];
@@ -5561,6 +5579,13 @@ void RenderingDeviceVulkan::_draw_list_insert_clear_region(DrawList *draw_list, 
 		} else {
 			ERR_CONTINUE(true);
 		}
+
+		// Ensure VkClearAttachment has been initialized properly.
+		ERR_CONTINUE_MSG(clear_at.aspectMask == VK_IMAGE_ASPECT_FLAG_BITS_MAX_ENUM ||
+								 clear_at.colorAttachment == uint32_t(-1) ||
+								 clear_at.clearValue.depthStencil.stencil == uint32_t(-1),
+				"Bug: VkClearAttachment not initialized properly.");
+
 		clear_attachments.push_back(clear_at);
 	}
 

